### ISSUE TITLE
Does not check for non equality of un-related classes

### ISSUE THREAD

**Issue by hcoles** [jqno/equalsverifier#131]
I believe you are familiar with pitest so I'll miss out the background on mutation testing.

Possibly I have missed something in the api, but based on the mutation coverage it looks like EqualsVerifier is not checking that equals implementations consider unrelated classes to be not equal.

For a my Pair class from https://github.com/NCR-CoDE/QuickTheories

``` java
public final class Pair<A, B> {
  public final A _1;
  public final B _2;

  private Pair(A _1, B _2) {
    this._1 = _1;
    this._2 = _2;
  }

  public static <A, B> Pair<A, B> of(A a, B b) {
    return new Pair<>(a, b);
  }

  @Override
  public int hashCode() {
    return Objects.hash(_1, _2);
  }

  @Override
  public boolean equals(Object obj) {
    if (this == obj) // <-- expected surviving mutant
      return true;
    if (obj == null)
      return false;
    if (getClass() != obj.getClass())
      return false;  // <-- unexpected surviving mutant
    Pair other = (Pair) obj;
    return Objects.equals(_1, other._1) &&
        Objects.equals(_2, other._2);
  }

}
```

The test

``` java
  @Test
  public void shouldObeyHashcodeEqualsContract() {
    EqualsVerifier.forClass(Pair.class).allFieldsShouldBeUsed().verify();
  }
```

Leaves surviving mutants as shown.

The first is expected as the if statement is purely a performance optimisation, but the second is not.

If I add an additional hand rolled test the mutant is killed.

``` java
  @Test
  public void shouldNotBeEqualToUnrelatedClass() {
    assertThat(Pair.of(1, 2)).isNotEqualTo("");
  }
```

I've not dug into this in any detail, but it appears that EqualsVerifier is not performing a similar check itself.

**Comment by jqno** [comment]
Hi,
I've reproduced this with PIT, and you're right. I don't know why I don't have such a test case, but I'll add it to the next release. I will try to get that out before christmas :).

Jan

**Comment by jqno** [comment]
I've just released version 1.7.6, which fixes this!
